cpp/Platform.Equality/Platform.Equality.Any.h

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#ifndef PLATFORM_EQUALITY_ANY_H
+#define PLATFORM_EQUALITY_ANY_H
+
+#include <any>
+#include <functional>
+#include <typeindex>
+#include <concepts>
+#include <stdexcept>
+
+namespace Platform::Equality
+{
+    template<class T>
+    inline auto __to_any_equal_visitor(auto func)
+    {
+        return std::pair<std::type_index, std::function<bool(std::any, std::any)>>
+        {
+                std::type_index(typeid(T)),
+                [func](std::any a, std::any b) -> std::size_t
+                {
+                    if constexpr (std::is_void_v<T>)
+                        return func();
+                    else
+                        return func(std::any_cast<T>(a), std::any_cast<T>(b));
+                }
+        };
+    }
+
+    #define BASE_VISITOR_REGISTER(Type) __to_any_equal_visitor<Type>([](Type a, Type b) {return a == b;})
+    #define VISITOR_REGISTER(Type, Func) __to_any_equal_visitor<Type>([](Type a, Type b) {return Func;})
```

Comment:
should it be?
```cpp
#define VISITOR_REGISTER(Type, Func) __to_any_equal_visitor<Type>(Func)
```

Reply:
If [this change](https://github.com/linksplatform/Equality/pull/2#discussion_r614754863) is applied, this macro can be deleted as there will be no usages.